content/blog/kallisti5/2021-03-22_feb_2021_survey_results.md

```diff
@@ -0,0 +1,138 @@
+
+## Which version of Haiku do you normally use?
+
+<img src="/files/survey_feb2021/1.png" alt="59.7% Stable, 40.3% Nightly"><br/>
+
+So far, not too suprising. A lot of people value the stable R1/Beta 2 release, but most prefer the nightly images. This makes sense as R1 / Beta 2 was released around 9 months ago.
+
+## What architecture do you normally use?
+
+<img src="/files/survey_feb2021/2.png" alt="85.5% x86_64 (64-bit), 14.5% x86_gcc2 (32-bit)"><br/>
+
+We continue to support the 32-bit version of Haiku for its BeOS binary compatibility. However, it appears the trend continues to be users gravitating to the more modern x86_64.
+This further solidifies the idea that after R1, the 32-bit version of Haiku will be eventually deprecated.
+
+## How often do you use haiku on a virtual machine vs physical hardware?
+
+<img src="/files/survey_feb2021/3.png" alt="41.3% of people never use Haiku in a VM. 32.3% of people never use haiku on physical hardware"><br/>
+
+This one is interesting. It seems more users prefer running Haiku on physical hardware. Those who run Haiku on physical hardware also run Haiku daily more often then users who run Haiku in a virtual machine.
+
+## What is the most "common problem" you see when using Haiku?
+
+<img src="/files/survey_feb2021/4.png" alt="53.7% Hardware/Driver support, 37.7% Lack of software"><br/>
+
+This one doesn't give a lot of interesting data. Hardware / Driver support in the top position isn't suprising. Haiku supports a large amount of hardware natively for our size... however we definitely stagger behind Linux and the BSD's on hardware support.
+
+## What class of hardware has the most problems in your experience
+
+<img src="/files/survey_feb2021/5.png" alt="45.5% Graphics, 19.6% Network, "><br/>
+
+Not suprising.  While Haiku supports quite a few mainstream graphics cards, we are always finding edge cases and broken modesetting.
+There are some long-planned projects trying to piggyback on Linux's graphics drivers... however nothing has even really taken shape beyond our native drivers.
+
+Network is actually a little suprising to me here. I personally don't see too many issues with network cards.
+I'm wondering if by 'Network' users actually mean our control panel / wifi network selection / etc?
+
+## What class of software is the most lacking?
+
+<img src="/files/survey_feb2021/5.png" alt="60.2% Web Browser"><br/>
+
+Not suprising at all.<br/><br/>
+The web browser has catapulted to the "most important" application in the last decade for pretty much all tasks. (checking email, news, char, multimedia, some gaming)<br/><br/>
+WebPositive is based on the modern Webkit engine, however our fork of Webkit moves slowly on the tail of the *rapidly* moving Webkit project. We spend a lot of time rebasing our OS support on the latest Webkit changes. We hope to eventually get our OS support code merged so we can refocus on improving and stabilizing WebPositive over "keeping up"
```

Comment:
The main problem is all the work going on on webkit since beta2 was not made available to users, due to using changes in the OS that will only be available in beta3.
This slows down the response time when users report bugs on the web browser, and probably reduces the motivation for people contributing to it as their changes take a long time to become available.

Reply:
Don't take this as any kind of attack. You have been doing an amazing job with what you have to work with.
Do you have any concrete recommendations on how to improve the above statement?   Overall we saw a *TON* of people mentioning WebPositive (and the selection of browsers in general), so we should really hit the issue head on.

Reply:
Saying that our fork moves slowly is not true. The fork is usually maybe a month behind upstream, which is ok. The problem is mainly in getting this work in the hands of our users, and it should be solved by the beta3 release.

```suggestion
WebPositive is based on the modern Webkit engine, however, WebKit relies on a rater large amount of platform specific code. On one side, this makes sure it integrates very well with the OS (using, for example, the same font rendering as other applications). On the other, updating WebKit requires some development on our side and continues to push it to improve our APIs (for both network and drawing code, as well as other parts of the system). This also resulted on WebKit development waiting on the availability of a new stable release of Haiku (including new features) in order to ship updates. Recently we have worked on improving the ABI stability of some parts of Haiku to make it easier to ship improved versions of WebKit more often.
```

Reply:
Applied change to latest update.